src/herder/test/TransactionQueueTests.cpp

```diff
@@ -2167,14 +2208,26 @@ TEST_CASE("transaction queue with fee-bump", "[herder][transactionqueue]")
             TransactionQueueTest test{queue};
             auto tx1 = transaction(*app, account1, 1, 1, 100, 1, isSoroban);
             uint32_t discount = tx1->getFullFee() - tx1->getInclusionFee();
+            auto newInclusionToPay = 200;
+            if (isSoroban)
+            {
+                // In case of Soroban, provide additional discount to test the
+                // case where inclusion fee is less than balance, but total fee
+                // is not.
+                discount += newInclusionToPay;
```

Comment:
What do you mean here exactly? The test was already testing the right scenarios (not enough balance to cover the fee), but because it used only classic transactions, it did not check that there's enough balance to cover full fee, not just inclusion. So the setup changed a bit to make sure with Soroban, the account has enough balance to cover inclusion fee, but not full fee.

Reply:
In general I'd expect that if test setup changes, then either expectations, or covered logic changes. I guess it's ok that it still passes, but maybe there is some additional assertion that would fail before this change, but would pass after adding it.

Reply:
the setup for classic didn't actually change (submit first tx, such that available balance left is 1, then submit another tx and fail due to insufficient balance), hence I'm not expecting any new assertions to fail. a new setup for soroban was added (to produce the same failures but with enough balance to cover just the inclusion fee) but the expected behavior is the same, hence those asserts didn't change

Reply:
Hmm, I thought Soroban setup has already been there before this PR, which is why I got confused by this change.  If Soroban scenario hasn't been covered before, then this makes sense to me.

Reply:
oh, I see the confusion now. there were soroban scenarios indeed, but they were too strict, as they didn't provide enough balance to cover the inclusion fee. The updates introduced new scenarios, and changed the setup to cover the inclusion fee, but fail to cover the full fee (to test for a bug we fixed recently). The failure mode is the same (insufficient fee), which is why the test cases themselves didn't change.